src/main/java/uk/co/ramp/statistics/StatisticsRecorderImpl.java

```diff
@@ -0,0 +1,66 @@
+package uk.co.ramp.statistics;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.springframework.stereotype.Service;
+import uk.co.ramp.people.Case;
+import uk.co.ramp.statistics.types.ImmutableInfection;
+import uk.co.ramp.statistics.types.Infection;
+
+@Service
+public class StatisticsRecorderImpl implements StatisticsRecorder {
+
+  private final Map<Integer, Integer> personDaysIsolation = new HashMap<>();
+  private final Map<Integer, Integer> peopleInfected = new HashMap<>();
+  private final Map<Integer, Integer> contactsTraced = new HashMap<>();
+  private final Map<Integer, List<Infection>> r0Progression = new HashMap<>();
+
+  public Map<Integer, Integer> getContactsTraced() {
+    return contactsTraced;
+  }
+
+  public Map<Integer, Integer> getPeopleInfected() {
+    return peopleInfected;
+  }
+
+  public Map<Integer, Integer> getPersonDaysIsolation() {
+    return personDaysIsolation;
+  }
+
+  public Map<Integer, List<Infection>> getR0Progression() {
+    return r0Progression;
+  }
```

Comment:
If we do that, then the StatisticsWriter class needs the Impl rather than the interface, which would also affect the Context... unless I'm missing something... its been a long day so thats possible.